File: app/src/main/java/dev/ragnarok/fenrir/api/model/response/NewsfeedSearchResponse.java
package dev.ragnarok.fenrir.api.model.response;

import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import dev.ragnarok.fenrir.api.model.VKApiCommunity;
import dev.ragnarok.fenrir.api.model.VKApiPost;
import dev.ragnarok.fenrir.api.model.VKApiUser;

public class NewsfeedSearchResponse {
    @Nullable
    @SerializedName("items")
    public List<VKApiPost> items;

    @Nullable
    @SerializedName("profiles")
    public List<VKApiUser> profiles;

    @Nullable
    @SerializedName("groups")
    public List<VKApiCommunity> groups;

    @Nullable
    @SerializedName("next_from")
    public String nextFrom;

    //@SerializedName("count")
    //public Integer count;

    //@SerializedName("total_count")
    //public Integer totalCount;
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/response/NewsfeedSearchResponse.kt
package dev.ragnarok.fenrir.api.model.response

import dev.ragnarok.fenrir.api.model.VKApiCommunity
import dev.ragnarok.fenrir.api.model.VKApiPost
import dev.ragnarok.fenrir.api.model.VKApiUser
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class NewsfeedSearchResponse {
    @SerialName("items")
    var items: List<VKApiPost>? = null

    @SerialName("profiles")
    var profiles: List<VKApiUser>? = null

    @SerialName("groups")
    var groups: List<VKApiCommunity>? = null

    @SerialName("next_from")
    var nextFrom: String? = null //@SerialName("count")
    //public Integer count;
    //@SerialName("total_count")
    //public Integer totalCount;
}